recipe/meta.yaml

```diff
@@ -10,16 +10,16 @@ source:
   sha256: a14c1c5143081701d4fe62283e09cc5ad5336dff3a152de64eb77ee3180e9792
 
 build:
-  number: 1
+  number: 2
   noarch: python
 
 requirements:
   host:
     - python >=3.8
   run:
     - python >=3.8
-    - dask-core ={{ version }},!={{ version }}a*
-    - distributed ={{ version }},!={{ version }}a*
+    - dask-core {{ version }}.*,!={{ version }}a.*
+    - distributed {{ version }}.*,!={{ version }}a.*
```

Comment:
Am still a little worried about adding `.`s here not behaving correctly. From the Conda-Builds docs on `run`:

> Package names should follow the [package match specifications]( https://conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#package-match-specifications ).

From [the match spec docs]( https://conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#package-match-specifications ):

> \* matches 0 or more characters in the version string. In terms of regular expressions, it is the same as r'.*'.

> The fuzzy constraint numpy=1.11 matches 1.11, 1.11.0, 1.11.1, 1.11.2, 1.11.18, and so on.

So the previous syntax should already be doing what we want. The fact it isn't is a little strange and may indicate a bug.

The use of `.`s shouldn't be needed and may lead to incorrect behavior.

Given all this, am wondering if the `,` is what is causing us issues

Reply:
I think the comma could be what's causing the issue here - I noticed that from the conflict message I shared in https://github.com/conda-forge/dask-feedstock/pull/190#issuecomment-1215624680 that a `.*` seemingly got appended to the end of our constraint automatically.

I'm not sure if this is documented anywhere, but I would assume it gets done to map fuzzy constraints from the `conda install` style to the `conda build` style, i.e.

```
numpy=1.11 -> numpy 1.11.*
```

but might not be expecting there to be two constraints in one line and so ends up behaving strangely.

I'll try out your suggestions and see if those get the intended behavior - out of interest, is there a way to inspect the dependencies of a locally built conda package? Trying `mamba repoquery depends` but not sure if it has support for local packages